### ISSUE TITLE
[Bug]: internal/tray is rendered with global margins when empty

### ISSUE THREAD

**Issue by agorgl** [polybar/polybar#3036]
### Checklist

- [X] I have read the appropriate section in the [contributing guidelines](https://github.com/polybar/polybar/blob/master/CONTRIBUTING.md)
- [X] I believe this issue is a problem with polybar itself and not a misconfiguration on my part
- [X] I have searched for other open and closed [issues](https://github.com/polybar/polybar/issues?q=is%3Aissue) that may have already reported this problem
- [X] I have checked the [known issues](https://github.com/polybar/polybar/wiki/Known-Issues) page for this problem.
- [X] I have followed the [debugging guide](https://github.com/polybar/polybar/wiki/Debugging-your-Config) to narrow down the problem to a minimal config.

### Steps to reproduce

```
polybar -r main
```

### Minimal config

```dosini
[bar/main]
width = 100%
module-margin-left = 2
module-margin-right = 0
modules-right = time tray

[module/tray]
type = internal/tray
format-offset = 0px
format-margin = 0px
format-padding = 0px
tray-padding = 0px
tray-spacing = 8px
```

### Expected behavior

When tray is empty (no active items in tray) rendering it along with the global margins `module-margin-left` and `module-margin-right` should be skipped to avoid creating unnecessary empty space, I believe this is handled here:

https://github.com/polybar/polybar/blob/7d7bf84e16f19c31cc89ac4406224f864cb2fb58/src/components/controller.cpp#L519C44-L519C44

### Actual behavior

The empty tray is rendered along with the global margins

### Window Manager and Version

bspwm

### Linux Distribution

Arch

### Polybar version

```text
polybar 3.7.0

Features: +alsa +curl +i3 +mpd +network(libnl) +pulseaudio +xkeyboard

X extensions: +randr (+monitors) +composite +xkb +xrm +xcursor

Build type: Release
Compiler: /usr/bin/c++
Compiler flags: -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection -Wp,-D_GLIBCXX_ASSERTIONS -g -ffile-prefix-map=/build/polybar/src=/usr/src/debug/polybar -flto=auto -O3 -DNDEBUG -Wall -Wextra -Wpedantic -Wdeprecated-copy-dtor -Wsuggest-override
Linker flags: -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -flto=auto -Wall -Wextra -Wpedantic -Wdeprecated-copy-dtor -Wsuggest-override  -Wall -Wextra -Wpedantic -Wdeprecated-copy-dtor -Wsuggest-override
```

### Additional Context / Screenshots

Expected (behavior of legacy tray):
![image](https://github.com/polybar/polybar/assets/4622346/2e44ff5d-55e1-4d87-a95a-3d5565d5be20)

Actual (new internal/tray module):
![image](https://github.com/polybar/polybar/assets/4622346/dc429c80-5ddc-4f0a-84e3-4ac20f657411)

I suppose this could be tackled in two possible ways:
1. mark internal/tray module as not visible when empty and avoid rendering it in the main modules loop
2. allow overriding `module-margin-{left,right}` on a per module basis (would give additional flexibility to polybar in general)

**Comment by patrick96** [comment]
Good catch!

Normally, modules that don't produce output already don't have the module margin (and separator) applied.

But the tray module produces some formatting tags that are interpreted by the renderer to produce the tray icons. The problem is that those tags are always there, so the module doesn't look empty.

The fix is not as straightforward as I thought, but this should be solveable.

**Comment by patrick96** [comment]
I have fixed this in #3037 and it should be released as version 3.7.1 soon.

**Comment by agorgl** [comment]
Nice, thank you!
